tune_sklearn/tune_search.py

```diff
@@ -656,17 +673,29 @@ def _tune_run(self, config, resources_per_trial):
                 override_search_space = False
 
             search_kwargs = self.search_kwargs.copy()
-            search_kwargs.update(metric=self._metric_name, mode="max")
+            if override_search_space:
+                search_kwargs["metric"] = run_args.pop("metric")
+                search_kwargs["mode"] = run_args.pop("mode")
+                if run_args["scheduler"]:
+                    if hasattr(run_args["scheduler"], "_metric") and hasattr(
+                            run_args["scheduler"], "_mode"):
+                        run_args["scheduler"]._metric = search_kwargs["metric"]
+                        run_args["scheduler"]._mode = search_kwargs["mode"]
+                    else:
+                        warnings.warn(
+                            "Could not set `_metric` and `_mode` attributes "
+                            f"on Scheduler {run_args['scheduler']}. "
+                            "This may cause an exception later! "
+                            "Ensure your Scheduler initializes with those "
+                            "attributes.", UserWarning)
```

Comment:
is there some way of setting this, like with `set_search_properties`?

Reply:
We can't use `set_search_properties` as that would cause issues when it's passed to `tune.run`, which calls it itself - and it expects an object which hadn't had it called before